Userland/Libraries/LibWeb/HTML/HTMLScriptElement.cpp

```diff
@@ -49,236 +50,264 @@ void HTMLScriptElement::begin_delaying_document_load_event(DOM::Document& docume
+    if (!has_attribute(HTML::AttributeNames::src) && source_text.is_empty()) {
         return;
+    }
 
-    // 7. If the element is not connected, then return. The script is not executed.
+    // 7. If el is not connected, then return.
     if (!is_connected()) {
         dbgln("HTMLScriptElement: Refusing to run script because the element is not connected.");
         return;
     }
 
-    // 8. If either:
-    //    - the script element has a type attribute and its value is the empty string, or
-    //    - the script element has no type attribute but it has a language attribute and that attribute's value is the empty string, or
-    //    - the script element has neither a type attribute nor a language attribute, then
-    //        ...let the script block's type string for this script element be "text/javascript".
+    // 8. If any of the following are true:
+    //    - el has a type attribute whose value is the empty string;
+    //    - el has no type attribute but it has a language attribute and that attribute's value is the empty string; or
+    //    - el has neither a type attribute nor a language attribute
     String script_block_type;
-    bool has_type = has_attribute(HTML::AttributeNames::type);
-    bool has_language = has_attribute(HTML::AttributeNames::language);
-    if ((has_type && attribute(HTML::AttributeNames::type).is_empty())
-        || (!has_type && has_language && attribute(HTML::AttributeNames::language).is_empty())
-        || (!has_type && !has_language)) {
+    bool has_type_attribute = has_attribute(HTML::AttributeNames::type);
+    bool has_language_attribute = has_attribute(HTML::AttributeNames::language);
+    if ((has_type_attribute && attribute(HTML::AttributeNames::type).is_empty())
+        // then let the script block's type string for this script element be "text/javascript".
+        || (!has_type_attribute && has_language_attribute && attribute(HTML::AttributeNames::language).is_empty())
+        || (!has_type_attribute && !has_language_attribute)) {
         script_block_type = "text/javascript";
-    } else if (has_type) {
-        // Otherwise, if the script element has a type attribute, let the script block's type string for this script element be the value of that attribute.
-        script_block_type = attribute(HTML::AttributeNames::type);
-    } else if (!attribute(HTML::AttributeNames::language).is_empty()) {
-        // Otherwise, the element has a non-empty language attribute; let the script block's type string for this script element be the concatenation of the string "text/" followed by the value of the language attribute.
+    }
+    // Otherwise, if el has a type attribute,
+    else if (has_type_attribute) {
+        // then let the script block's type string be the value of that attribute with leading and trailing ASCII whitespace stripped.
+        script_block_type = attribute(HTML::AttributeNames::type).trim_whitespace();
```

Comment:
Linus added some fancy helpers to LibWeb/Infra for things like `https://infra.spec.whatwg.org/#strip-leading-and-trailing-ascii-whitespace` which is used below on line 180 `script_block_type.trim(Infra::ASCII_WHITESPACE)` which should probably be used here for consistency